app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -235,8 +237,22 @@ open class AccountStore(
 
         webView.clearCache(true)
         clearLogs()
+        removeDeviceFromFxA()
+    }
+
+    private fun removeDeviceFromFxA() {
+        if(fxa != null) {
+            val account = mozilla.appservices.fxaclient.FirefoxAccount.fromJSONString(fxa!!.toJSONString())
```

Comment:
Whoa, a lot going on here! We convert the `fxa` to JSON and then convert it back!

Why would we do that as opposed to just using the current FxA instance? I can see why things would get tripped up here because the calls to remove the previous stored instance and generate a new one occur _before_ we go through to remove the device, but that should be ameliorated by moving those calls after the `removeDeviceFromFxA` call (`this.securePreferences.remove(Constant.Key.firefoxAccount)` and `this.generateNewFirefoxAccount()` specifically). 

Also, checking to see if something is null usually means that we can use the `let` syntax. So this could read:
```kotlin
fxa?.let {
   val registeredDevices = it.getDevices()
    for (device in registeredDevices) {
        if(device.isCurrentDevice) {
             account.destroyDevice(device.id)
         }
    }
}
```
or even
```kotlin
val registeredDevices = fxa?.getDevices ?: []
for (device in registeredDevices) {
   if(device.isCurrentDevice) {
      account.destroyDevice(device.id)
   }
}
```

Reply:
We can't access the `getDevices` method from an a-c FxA account. It can only be retrieved from an a-s FxA client. The companion obj in the FxA client says to do the conversion from a-c using the json method: 
```
companion object {
        /**
         * Restores the account's authentication state from a JSON string produced by
         * [FirefoxAccount.toJSONString].
         *
         * This does not make network requests, and can be used on the main thread.
         *
         * @return [FirefoxAccount] representing the authentication state
         */
        fun fromJSONString(json: String, persistCallback: PersistCallback? = null): FirefoxAccount {
            return FirefoxAccount(rustCall { e ->
                LibFxAFFI.INSTANCE.fxa_from_json(json, e)
            }, persistCallback)
        }
    }
```

Reply:
https://github.com/mozilla/application-services/blob/master/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

Reply:
(I like the `let` syntax and I am going to add that)

Reply:
OK, then I think we should request that the android-components version of `FirefoxAccount` reflects the `application-services` implementation. They should be the same thing, or we should be using the `application-services` version everywhere.